src/pykiso/lib/connectors/cc_pcan_can.py

```diff
@@ -397,47 +398,110 @@ def _merge_trc(self) -> None:
+                data = trc.read().splitlines(True)
+
+                first_trc_start_time = CCPCanCan._get_trace_start_time(
+                    data, first_message_line
+                )
+                message_idx = len(data[first_message_line:])
+
+            list_of_traces.pop(0)
+
             # Append all trace files
             for file in list_of_traces:
                 with open(file, "r") as trc:
                     data = trc.read().splitlines(True)
+
+                    # Get offset between current and first trc file
+                    trc_start_time = CCPCanCan._get_trace_start_time(
+                        data, first_message_line
+                    )
+                    offset = trc_start_time - first_trc_start_time
+
+                    # Fix message number and time offset inconstistancies
+                    corrected_data = CCPCanCan._fix_merge_inconsistencies(
+                        data[first_message_line:], offset, message_idx
+                    )
+                    message_idx += len(data[first_message_line:])
+
                 with open(result_trace, "a") as merged_trc:
-                    merged_trc.writelines(data[first_message_line:])
+                    merged_trc.writelines(corrected_data)
                 os.remove(file)
 
-            self._fix_message_numbers(result_trace, first_message_line)
-
         except IndexError:
             log.internal_warning("No trace to merge")
 
-    def _fix_message_numbers(
-        self, merged_trace: Union[Path, str], first_message_line: int
-    ) -> None:
-        """Parse message numbers in the merged file and fix inconsistencies caused by the
-        merging process.
+    @staticmethod
+    def _fix_merge_inconsistencies(
+        data: List[str], offset: float, message_idx: int
+    ) -> List[str]:
+        """Fix merge inconsistencies such as message number and time offset
 
-        :param merged_trace: merged trace file to fix
-        :param first_message_line: line number of the first logged can message
+        :param data: can messages to fix
+        :param offset: time offset beetween first trace and current trace in milliseconds
+        :param message_idx: message number from last trace file
+
+        :return: corrected can messages
         """
-        message_nb_index = 7
-        with open(merged_trace, "r+") as trc:
-            merged_data_lines = trc.read().splitlines(True)
-
-            # Parsing to keep the message numbers consistent after the merge
-            for line_number in range(first_message_line, len(merged_data_lines)):
-                message_number = str((line_number + 1) - first_message_line)
-                merged_data_lines[line_number] = (
-                    message_number.rjust(message_nb_index)
-                    + merged_data_lines[line_number][message_nb_index:]
-                )
-            trc.seek(0)
-            trc.writelines(merged_data_lines)
+        message_number_idx = 7
```

Comment:
I think you need to explain where this numbers are coming from. Maybe in the description, provide a note showing why? How?

Reply:
It is the indexes for the parsing of a line, is it worth documenting that thought? As it will disappear when we switch to the trc reader